server/mscalendar/availability.go

```diff
@@ -63,9 +71,84 @@ func (m *mscalendar) SyncAll() (string, *StatusSyncJobSummary, error) {
 		return "", &StatusSyncJobSummary{}, errors.Wrap(err, "not able to load the users from user index")
 	}
 
+	err = m.Filter(withSuperuserClient)
```

Comment:
It's possible this could have side effects since this mutates state, and thus affects functions like `syncUsers`. From a quick look, it seems that it won't have issues though.

What's weird is that the only client used in that path is in `GetCalendarViews`, which is setting its own user-level client for batch requests. Not sure how that is working this way:

https://github.com/mattermost/mattermost-plugin-mscalendar/blob/d78381aa8bf00887bab751a9d2a33d7571c6c5e5/server/mscalendar/availability.go#L388-L389